tests/test_lit_server.py

```diff
@@ -120,6 +121,25 @@ async def test_stream(simple_stream_api):
             ), "Server returns input prompt and generated output which didn't match."
 
 
+@pytest.mark.asyncio()
+async def test_stream_client_disconnection(simple_stream_api, caplog):
```

Comment:
might need to slightly change the approach to testing. The task seems to get completed before it is cancelled.